### PR TITLE
[IOTDB-438] Modify SeriesReader to uniform usage

### PR DESCRIPTION
**Pull request by liutaohua**

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/query/reader/seriesRelated/SeriesDataPointReader.java [hunk 1]
```diff
@@ -18,37 +18,59 @@
...
+  private SeriesDataRandomReader randomReader;
   private boolean hasCachedTimeValuePair;
   private BatchData batchData;
   private TimeValuePair timeValuePair;
 
-  public SeriesReaderWithValueFilter(Path seriesPath, TSDataType dataType, Filter valueFilter,
-      QueryContext context, QueryDataSource dataSource) {
-    super(seriesPath, dataType, null, context, dataSource);
-    this.valueFilter = valueFilter;
+  public SeriesDataPointReader(Path seriesPath, TSDataType dataType, Filter timeFilter,
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/query/reader/seriesRelated/SeriesDataPointReader.java]:
```suggestion
  public SeriesDataPointReader(Path seriesPath, TSDataType dataType,
```

File: server/src/main/java/org/apache/iotdb/db/query/reader/seriesRelated/SeriesDataPointReader.java [hunk 2]
```diff
@@ -18,37 +18,59 @@
...
+      Filter valueFilter, QueryContext context, QueryDataSource dataSource) {
+    randomReader = new SeriesDataRandomReader(seriesPath, dataType, context,
+        dataSource.getSeqResources(), dataSource.getUnseqResources(), timeFilter, valueFilter);
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/query/reader/seriesRelated/SeriesDataPointReader.java]:
```suggestion
        dataSource.getSeqResources(), dataSource.getUnseqResources(), null, valueFilter);
```

File: server/src/main/java/org/apache/iotdb/db/query/reader/seriesRelated/SeriesDataRandomReader.java [hunk 3]
```diff
@@ -46,15 +47,18 @@
 import org.apache.iotdb.tsfile.read.controller.ChunkLoaderImpl;
 import org.apache.iotdb.tsfile.read.controller.IChunkLoader;
 import org.apache.iotdb.tsfile.read.filter.basic.Filter;
+import org.apache.iotdb.tsfile.read.filter.basic.UnaryFilter;
 import org.apache.iotdb.tsfile.read.reader.IChunkReader;
 import org.apache.iotdb.tsfile.read.reader.IPageReader;
 import org.apache.iotdb.tsfile.read.reader.chunk.ChunkReader;
 
-public abstract class AbstractSeriesReader {
+public class SeriesDataRandomReader implements IDataRandomReader {
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/query/reader/seriesRelated/SeriesDataRandomReader.java]:
```suggestion
public class EngineSeriesReader implements IEngineSeriesReader {
```

File: server/src/main/java/org/apache/iotdb/db/query/reader/seriesRelated/SeriesDataRandomReader.java [hunk 4]
```diff
@@ -80,20 +84,44 @@
...
-    return getFilter() == null
-        || getFilter().containStartEndTime(statistics.getStartTime(), statistics.getEndTime());
+    return timeFilter == null
+        || timeFilter.containStartEndTime(statistics.getStartTime(), statistics.getEndTime());
+  }
+
+  @Override
+  public boolean canUseCurrentChunkStatistics() {
+    Statistics chunkStatistics = currentChunkStatistics();
+    return !isChunkOverlapped() && satisfyFilter(chunkStatistics);
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/query/reader/seriesRelated/SeriesDataRandomReader.java]:
```suggestion
    return !isChunkOverlapped() && satisfyTimeFilter(chunkStatistics);
```

File: server/src/main/java/org/apache/iotdb/db/query/reader/seriesRelated/SeriesDataRandomReader.java [hunk 5]
```diff
@@ -80,20 +84,44 @@
...
 
   protected boolean satisfyFilter(Statistics statistics) {
-    return getFilter() == null
-        || getFilter().containStartEndTime(statistics.getStartTime(), statistics.getEndTime());
+    return timeFilter == null
+        || timeFilter.containStartEndTime(statistics.getStartTime(), statistics.getEndTime());
+  }
+
+  @Override
+  public boolean canUseCurrentChunkStatistics() {
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/query/reader/seriesRelated/SeriesDataRandomReader.java]: add javadoc, only be used for aggregate without value filter

File: server/src/test/java/org/apache/iotdb/db/engine/merge/MergeTaskTest.java [hunk 6]
```diff
@@ -36,7 +36,7 @@
 import org.apache.iotdb.db.exception.metadata.MetadataException;
 import org.apache.iotdb.db.exception.path.PathException;
 import org.apache.iotdb.db.query.context.QueryContext;
-import org.apache.iotdb.db.query.reader.seriesRelated.RawDataReaderWithoutValueFilter;
+import org.apache.iotdb.db.query.reader.seriesRelated.SeriesDataBatchReader;
```
Comment (qiaojialin) [inline server/src/test/java/org/apache/iotdb/db/engine/merge/MergeTaskTest.java]:
```suggestion
import org.apache.iotdb.db.query.reader.seriesRelated.SeriesRawDataReader;
```

File: server/src/main/java/org/apache/iotdb/db/query/reader/seriesRelated/SeriesDataBatchReader.java [hunk 7]
```diff
@@ -29,34 +29,27 @@
...
   private boolean hasRemaining;
   private boolean managedByQueryManager;
 
-  public RawDataReaderWithoutValueFilter(Path seriesPath, TSDataType dataType, Filter filter,
-      QueryContext context, QueryDataSource queryDataSource) {
-    super(seriesPath, dataType, context, queryDataSource.getSeqResources(),
-        queryDataSource.getUnseqResources());
-    this.filter = queryDataSource.setTTL(filter);
+  public SeriesDataBatchReader(Path seriesPath, TSDataType dataType, Filter filter,
+      Filter valueFilter, QueryContext context, QueryDataSource queryDataSource) {
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/query/reader/seriesRelated/SeriesDataBatchReader.java]:
```suggestion
      QueryContext context, QueryDataSource queryDataSource) {
```

File: server/src/main/java/org/apache/iotdb/db/query/reader/seriesRelated/SeriesDataBatchReader.java [hunk 8]
```diff
@@ -29,34 +29,27 @@
...
+    randomReader = new SeriesDataRandomReader(seriesPath, dataType, context,
+        queryDataSource, queryDataSource.setTTL(filter), valueFilter);
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/query/reader/seriesRelated/SeriesDataBatchReader.java]:
```suggestion
        queryDataSource, queryDataSource.setTTL(filter), null);
```

File: server/src/main/java/org/apache/iotdb/db/query/reader/seriesRelated/SeriesDataRandomReader.java [hunk 9]
```diff
@@ -419,4 +490,148 @@ public void close() throws IOException {
...
+      }
+      throw new IOException("no next batch");
+    }
+
+    @Override
+    public void close() throws IOException {
+    }
+  }
+
+  private class Ite implements IPointReader {
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/query/reader/seriesRelated/SeriesDataRandomReader.java]: SeriesPointReader

File: server/src/main/java/org/apache/iotdb/db/query/reader/seriesRelated/SeriesDataRandomReader.java [hunk 10]
```diff
@@ -419,4 +490,148 @@ public void close() throws IOException {
...
+
+  public IPointReader getIPointReader() {
+    return new Ite();
+  }
+
+  public IBatchReader getIBatchReader() {
+    return new BatchIte();
+  }
+
+  private class BatchIte implements IBatchReader {
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/query/reader/seriesRelated/SeriesDataRandomReader.java]: SeriesBatchReader

File: server/src/main/java/org/apache/iotdb/db/query/reader/seriesRelated/SeriesDataRandomReader.java [hunk 11]
```diff
@@ -419,4 +490,148 @@ public void close() throws IOException {
       openedChunkLoader.close();
     }
   }
+
+  public IPointReader getIPointReader() {
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/query/reader/seriesRelated/SeriesDataRandomReader.java]:
```suggestion
  public IPointReader getPointReader() {
```

File: server/src/main/java/org/apache/iotdb/db/query/reader/seriesRelated/SeriesDataRandomReader.java [hunk 12]
```diff
@@ -419,4 +490,148 @@ public void close() throws IOException {
...
+    return new Ite();
+  }
+
+  public IBatchReader getIBatchReader() {
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/query/reader/seriesRelated/SeriesDataRandomReader.java]:
```suggestion
  public IBatchReader getBatchReader() {
```

File: server/src/main/java/org/apache/iotdb/db/query/reader/seriesRelated/SeriesDataRandomReader.java (hunk 5 above)
Comment (liutaohua) [reply server/src/main/java/org/apache/iotdb/db/query/reader/seriesRelated/SeriesDataRandomReader.java]: done

**Pull request merged**